### ISSUE TITLE
Consider reducing Mac usage

### ISSUE THREAD

**Issue by BanzaiMan** [catapult-project/perfetto#1]
Currently, this repository is creating 5 Mac jobs for each build. With active development and cron jobs, it has more than 300 jobs queued up (but none of them is running; they are just waiting).

This is a huge burden on the system. Please consider canceling old ones and reducing the load.

Thanks.

**Comment by BanzaiMan** [comment]
Sorry for the noise; we'll review the issue on our side.

**Comment by primiano** [comment]
Hey @BanzaiMan , yeah I noticed that there is some problem with the mac bots. It is pretty evident once you look to our CI frontend: https://perfetto-ci.appspot.com/

Here's a couple of things on my side:

1) I think there is something odd with the mac builders. Our CI takes some minutes to build and <1 minute to run. [This](https://travis-ci.org/catapult-project/perfetto/jobs/330308613) is a recent job I see in our queue and it took 5 mins to run. Can't tell why they are piling up.

2) Yes, we kick 4 jobs for every CL sent. I would hope that your infrastructure isolates project, so even if we tick 1M jobs we don't end up stealing capacity from other projects.

3) I'd also hope that you have some sort of TTL and give up on jobs > 1 day older (that would be definitely acceptable for us).

4) As a stopgap solution, given that mac is best-effort support right now, I would be fine if we did exclude mac jobs from any branch other than master. If you give me an option to kick mac builds only for master I'll tick it (but only for mac, we really need jobs on all branches on the Linux builders)

Feel free to contact me, either here over email for more details.

P.S. FWIW we are paying Travis to get dedicated quota, but I suspect you realized that later on after filing the bug ;-)
